### PR TITLE
Removed dredd, removed unused HTTP methods from API, added unit tests

### LINKED ISSUE #83: Tests are broken on OSX
They break on the dredd step. Not sure why. Happy to help someone debug, but I don't have a mac. If you want to pair on it, let me know.

### LINKED ISSUE #121: Remove dredd
The dredd test started failing without any changes on our end, even with up being pinned to a version of dredd. This testing is just too much of a hassle to be worth the maintenance. We should remove it and replace it with unit tests covering the same things.

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?

Fixes #121 
Fixes #83
#### What's this PR do?
- Removes dredd and related requirements like node.js
- Disables unused methods like PATCH, PUT, DELETE on course and module detail views, and POST on module list view.
- Adds unit tests to cover module and course APIs for GET methods.
#### Where should the reviewer start?

courses/views.py
#### How should this be manually tested?

Just run the unit tests locally. We might want to do this on a Mac as well to confirm #83 is fixed

### CODE REVIEW

File: courses/views_test.py [hunk 1]
```diff
@@ -90,6 +106,79 @@ def test_create_course_accepts_instructors(self):
...
         assert EdxAuthor.objects.count() == 2
 
+    def test_get_course_list(self):
+        """
+        The course list API should list all courses available.
+        """
+        course = CourseFactory.create()
+        resp = self.client.get(reverse('course-list'))
+        assert resp.status_code == 200, resp.content.decode('utf-8')
+        course_list = json.loads(resp.content.decode('utf-8'))
```
Comment (justinabrahms) [inline courses/views_test.py:116]: `resp.json` should work, given this is django 1.9 https://docs.djangoproject.com/en/1.9/topics/testing/tools/#django.test.Response.json

File: lint.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ function incr_err() {
 }
 
 # check that mock.patch calls use autospec
-OUTPUT=`git grep "patch(" | grep -v "autospec"`
+OUTPUT=`git grep "mock.patch(" | grep -v "autospec"`
```
Comment (justinabrahms) [inline lint.sh]: Can we support this either being `mock.patch` or `patch()` with no preceeding thing?

This no longer catches the case wherein you do:

``` python
from mock import patch
patch(...)
```

**Comment by justinabrahms** [comment]
:+1:

File: courses/views_test.py (hunk 1 above)
Comment (noisecapella) [inline courses/views_test.py:116]: `resp` here is actually of type `rest_framework.response.Response` which ironically does not have the `json` method

**Comment by honzajavorek** [comment]
@justinabrahms @noisecapella 

Please note the latest version of Dredd (v1.0.7) fixes the problems you had with it. I see you removed Dredd from your build due to that :cry: If you mind to give it a chance, it would be awesome if you could confirm whether it now works correctly for you. I'll close the https://github.com/apiaryio/dredd/issues/348 at this moment, but if you find any issues again, please report back, re-open and let us know. Thanks!

**Pull request merged**